docs/internals/delete.md

```diff
@@ -0,0 +1,56 @@
+# Resource Deletion
+
+Resource deletion can vary depending on under which scenario deletion is
+occurring. At a high level a resource can be deleted:
+
+- `RegisterResource` it not called for an existing resource.
+- `pulumi dn` is called to destroy the resource
+- During `pulumi up` `Diff` is called and reports a resource replacement. The
+  replacement can either have:
+  - DeleteBeforeReplace=true
+  - DeleteBeforeReplace=false
+
+## RegisterResource not called for existing resource
+
+In this case there is a resource that was previously created by the TF Module
+that is now not being created. As part of the `ModuleComponent` the `TF Apply`
+will be run and the updated state will be returned which will not contain the
+removed resource. The ModuleComponent will then not emit a `RegisterResource`
+call which will cause the Pulumi engine to call `Delete` on the child resource.
+
+- The `Delete` call will happen in a different provider process so it will not
```

Comment:
This really is so surprising. I wonder if this is worth tracking as a Pulumi bug.